src/methods/nft/nft.ts

```diff
@@ -206,53 +220,229 @@ export const stageNewLibraryAsset = async (
+    attribs.push(createTextAttrib('location_type', locationType, false));
+    attribs.push(createTextAttrib('location', location, false));
+    attribs.push(createTextAttrib('content_type', contentType, false));
+    if (locationType === 'canister' && file) {
+      attribs.push(createTextAttrib('content_hash', getFileHash(file.rawFile), false));
+    }
+    attribs.push(createNatAttrib('size', size, false));
+    attribs.push(createNatAttrib('sort', maxSort + 1n, false));
+    attribs.push(createTextAttrib('read', 'public', false));
+    if (immutable) {
+      attribs.push(createBoolAttrib('com.origyn.immutable_library', true, true));
+    }
+
+    const libraryMetadata =  { Class: attribs };
+    return libraryMetadata;
+};
+
+export const stageCollectionLibraryAsset = async (
+  tokenId: string = '',
+  title: string = '',
+  collectionLibraryId: string,
+  immutable: boolean = false
+): Promise<OrigynResponse<any, StageLibraryAssetErrors | GetCollectionErrors>> => {
+  try {
+    const metadata = await buildLibraryMetadata(tokenId, collectionLibraryId, title, 'collection', immutable);
+
+    const libraryAsset: LibraryFile = {
+      library_id: collectionLibraryId,
+      library_file: { filename:'', path: '', size: 0, rawFile: Buffer.from([]) },
+    };
+
+    const metrics: Metrics = { totalFileSize: 0 };
+
+    return await canisterStageLibraryAsset(libraryAsset, tokenId, metrics, metadata);
+
+  } catch (err: any) {
+    return { err: { error_code: StageLibraryAssetErrors.ERROR_WHILE_STAGING, text: err?.message || err } };
+  }
+}
+
+export const stageWebLibraryAsset = async (
+  tokenId: string = '',
+  title: string = '',
+  webUrl: string,
+  immutable: boolean = false
+): Promise<OrigynResponse<any, StageLibraryAssetErrors | GetCollectionErrors>> => {
+
+  try {
+    const libraryId = title.toLowerCase().replace(/\s+/g, '-');
+
+    const metadata = await buildLibraryMetadata(tokenId, libraryId, title, 'web', immutable, undefined, webUrl);
+
+    const libraryAsset: LibraryFile = {
+      library_id: libraryId,
+      library_file: { filename:'', path: '', size: 0, rawFile: Buffer.from([]) },
+    };
+
+    const metrics: Metrics = { totalFileSize: 0 };
+
+    return await canisterStageLibraryAsset(libraryAsset, tokenId, metrics, metadata);
+
+  } catch (err: any) {
+    return { err: { error_code: StageLibraryAssetErrors.ERROR_WHILE_STAGING, text: err?.message || err } };
+  }
+}
+
 export const stageLibraryAsset = async (
-  files: StageFile[],
-  token_id?: string,
-): Promise<OrigynResponse<any, StageLibraryAssetErrors | GetCollectionErrors | GetNftErrors>> => {
+  file: StageFile,
+  tokenId: string = '',
+  title: string = '',
+  immutable: boolean = false
+): Promise<OrigynResponse<any, StageLibraryAssetErrors>> => {
+
   try {
+
     // Get the Raw file if called from a node context (csm.js)
     // tslint:disable-next-line prefer-for-of
-    for (let i = 0; i < files.length; i++) {
```

Comment:
Because the additional arguments would have to be arrays as well with each index corresponding to the files. We could add title and immutable to the StageFile type however. Would that work?